pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -346,26 +346,33 @@ func (p *PoolManager) isRelatedToKubevirt(pod *corev1.Pod) bool {
 	return false
 }
 
-func (p *PoolManager) releaseMacAddressesFromInterfaceMap(allocations map[string]string) {
-	for _, value := range allocations {
-		delete(p.macPoolMap, value)
+// changes mac's macPoolMap status to be deleted in controller context.
+func (p *PoolManager) MarkMacAddressesForRelease(allocations map[string]string) {
+	for _, macAddress := range allocations {
+		if _, exist := p.macPoolMap[macAddress]; p.IsMacInRange(macAddress) && exist {
```

Comment:
Is it necessary to check if a mac is in the range if it is already in the macPoolMap?

Reply:
yes. The mac updated can possibly be manually inserted, and as such - can also be out of range. 
since the webhook/controller only handle Macs that belong to the kmp-mmgr pod's range - then we need to check if it is in range before inserting/deleting